wfqueue.c

```diff
@@ -313,11 +313,13 @@ static void help_deq(queue_t *q, handle_t *th, handle_t *ph) {
         for (; idx == old && new == 0; ++i) {
             cell_t *c = find_cell(&h, i, th);
 
-            long Di = q->Di;
```

Comment:
If here increase Di, and then return returned  BOT beat VAL, the VAL maybe lost.